### PR TITLE
update the color icon provider, and fix some regressions due to opt. const

### PR DESCRIPTION
**Pull request by devoncarew**
- update the colors icon provider to the latest from package:flutter
- fix some regressions in identifying in-line colors and icons due to optional `const`
- add tests for the editor annotation provider (for icons and colors)

### CODE REVIEW

File: src/io/flutter/editor/FlutterEditorAnnotator.java [hunk 1]
```diff
@@ -70,43 +63,77 @@ else if (element instanceof DartNewExpression) {
...
+          final int value = val;
           //noinspection UseJBColor
           final Color color = new Color((int)(value >> 16) & 0xFF, (int)(value >> 8) & 0xFF, (int)value & 0xFF, (int)(value >> 24) & 0xFF);
           attachColorIcon(element, holder, color);
         }
-        catch (NumberFormatException ignored) {
+      }
+    }
+    else if (element instanceof DartCallExpression) {
+      // IconData(0xe914)
```
[the remark this answers is not in the document]
Comment (devoncarew) [reply src/io/flutter/editor/FlutterEditorAnnotator.java]: Sure - will add a comment. The IntelliJ parser can't distinguish between call expressions, and call expressions that are really constructor calls.

File: testSrc/unit/io/flutter/editor/FlutterEditorAnnotatorTest.java [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+      final Annotation annotation = annotationHolder.get(0);
+      assertEquals(HighlightSeverity.INFORMATION, annotation.getSeverity());
+    });
+  }
+
+  @Test
+  public void locatesColorCtor() throws Exception {
+    run(() -> {
+      final PsiElement testIdentifier = setUpDartElement("main() { Color(0xFFE3F2FD); }", "Color", LeafPsiElement.class);
```
Comment (jacob314) [inline testSrc/unit/io/flutter/editor/FlutterEditorAnnotatorTest.java]: Could be worth it to add matching cases with
`const Color(0xFFE3F2FD)`

`const [Color(0xFFE3F2FD])`
and 
`new Color(0xFFE3F2FD)`
and
cases with excess whitespace
`new    Color   (
0xFFE3F2FD
)`
to make sure that the matching based on whitespace differences isn't fragile.

File: src/io/flutter/editor/FlutterEditorAnnotator.java [hunk 3]
```diff
@@ -20,23 +20,15 @@
...
  */
 public class FlutterEditorAnnotator implements Annotator {
-  private static final Logger LOG = Logger.getInstance(FlutterEditorAnnotator.class);
-
   @Override
   public void annotate(@NotNull PsiElement element, @NotNull AnnotationHolder holder) {
     if (holder.isBatchMode()) return;
 
     if (element instanceof DartReferenceExpression || element instanceof DartArrayAccessExpression) {
-      if (!isInFlutterModule(element)) {
```
[the remark this answers is not in the document]
Comment (devoncarew) [reply src/io/flutter/editor/FlutterEditorAnnotator.java]: I was worried about the expense here, which is why I removed it. Not having this check will mean that Flutter icons, and material colors, will show up in the editor margins of all Dart files - those from flutter projects as well as regular dart. I _think_ that's generally ok.

screen shot 2018-11-07 at 10 34 37 am

File: testSrc/unit/io/flutter/editor/FlutterEditorAnnotatorTest.java (hunk 2 above)
Comment (devoncarew) [reply testSrc/unit/io/flutter/editor/FlutterEditorAnnotatorTest.java]: Adding `const [Color(0xFFE3F2FD])` and `const Color ( 0xFFE3F2FD )` (which did catch an issue). The `new` cases are currently not supported, and, I'm not 100% sure how important it is to support them (most or all user references are of the form `Color.fooColor`; the const Color(0xFFF)` references are generally just in `package:flutter/material/colors.dart`).

**Pull request merged**